### PR TITLE
Dev to Main

### PR DESCRIPTION
**Pull request by ParinLL**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	"github.com/ParinLL/cyberpower-ups-exporter/collector"
+	"github.com/ParinLL/cyberpower-ups-exporter/logger"
+)
+
+func main() {
+	log, err := logger.Initialize()
+	if err != nil {
+		panic("can't initialize zap logger: " + err.Error())
+	}
+	defer log.Sync()
```
Comment (Jyny) [inline main.go:20]: missing err check

File: collector/collector.go [hunk 2]
```diff
@@ -204,67 +222,41 @@ func (collector *upsCollector) Collect(ch chan<- prometheus.Metric) {
 
 		switch variable.Name {
 		case ".1.3.6.1.4.1.3808.1.1.1.2.1.1.0":
-			ch <- prometheus.MustNewConstMetric(collector.batteryStatus, prometheus.GaugeValue, value)
+			ch <- prometheus.MustNewConstMetric(c.batteryStatus, prometheus.GaugeValue, value)
```
Comment (Jyny) [inline collector/collector.go:225]: do not use Must* func before check anything

File: main.go (hunk 1 above)
Comment (Jyny) [inline main.go:20]: missing err check

File: collector/collector.go (hunk 2 above)
Comment (Jyny) [inline collector/collector.go]: do not use Must* func before check anything

File: main.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		log.Fatal("Failed to create UPS collector", zap.Error(err))
+	}
+
+	prometheus.MustRegister(upsCollector)
+
+	http.Handle("/metrics", logger.LoggingMiddleware(log, promhttp.Handler()))
+
+	addr := getPort()
+	log.Info("Beginning to serve", zap.String("port", addr))
+	if err := http.ListenAndServe(addr, nil); err != nil {
```
Comment (Jyny) [inline main.go:38]: add graceful shutdown with os signal trap

**Review by Jyny — CHANGES_REQUESTED** [review]
overall LGTM
use lib with more stable way for production ready

**Comment by ParinLL** [comment]
Merge for First Release

**Pull request merged**